backend/models/category.py

```diff
@@ -0,0 +1,37 @@
+from sqlalchemy import Column, String
+from sqlalchemy.orm import validates
+from backend.models.base_model import BaseModel
+
+
+class Category(BaseModel):
+    __tablename__ = 'category'
+    name = Column('name', String(length=100), nullable=False)
+    description = Column('description', String(length=250), nullable=False)
+
+    def __init__(self, name: str, description: str) -> None:
+        self.name = name
+        self.description = description
+
+    @validates('name')
+    def validate_name(self, key, name):
+        if not name:
+            raise ValueError("Name is required")
+        if not isinstance(name, str):
+            raise ValueError("Name must be a string")
+        if not name.strip():
+            raise ValueError("Name can't be empty")
```

Comment:
Acho que o not name e not name.strip() é a mesma coisa, não é?

Reply:
Se for uma string só com espaços em branco não entra no `not name`

Reply:
Mas a name.strip() testaria tudo, não?